Sources/Butter/Butter.swift

```diff
@@ -7,18 +7,24 @@ import Foundation
 public typealias NetworkCompletion<T: Decodable> = (Result<T, Error>) -> ()
 
 // MARK - Injection Point for URLSession
-public protocol URLSessionDataTaskInterface {
+protocol URLSessionDataTaskInterface {
 	func dataTask(with request: URLRequest, completionHandler: @escaping (Data?, URLResponse?, Error?) -> Void) -> URLSessionDataTask
 }
```

Comment:
what's the purpose of the protocol ? just wondering since you can already pass in a URLSession via your other init